traffic_router/core/src/main/java/com/comcast/cdn/traffic_control/traffic_router/core/router/TrafficRouter.java

```diff
@@ -537,8 +542,12 @@ public HTTPRouteResult multiRoute(final HTTPRequest request, final Track track)
 			final List<Cache> caches = selectCaches(request, ds, track);
 
 			if (caches != null && !caches.isEmpty()) {
+				if (isClientSteeringDiversityEnabled() && caches.size() > selectedCaches.size()) {
+					caches.removeAll(selectedCaches);
```

Comment:
Is it safe to assume that the same cache won't appear in the 'caches' list more than once? If so then it seems like you could end up with an empty 'caches' list here.

Reply:
Yes, I believe that is safe to assume. `selectCaches` should never return a list with duplicate caches.